### ISSUE TITLE
Support PipeWire

### ISSUE THREAD

**Issue by kaixiong** [Libvisual/libvisual#264]
PipeWire is becoming a standard on Linux and we should have an input plugin supporting it.

**Comment by hartwork** [comment]
@kaixiong I agree and already played a bit with client code but it's a bit of bumpy road, and https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2997 only is the start of it. If it was less bumpy I had already added a version to my https://github.com/hartwork/audio_pump_demo . Note that their API seems to be fully asynchronous, so we'll need tricks similar to what we have for PulseAudio's synchronous API. Happy to learn I'm wrong about any of this :smiley:

**Comment by hartwork** [comment]
PS: What I find interesting is how they hand out and take in buffers with `pw_stream_dequeue_buffer` and `pw_stream_queue_buffer`.
